content/blog/tech/native-application-or-pwa.md

```diff
@@ -0,0 +1,75 @@
+
+!!! note "Un code source pour les gouverner toutes"
+    Depuis quelques années, l'utilisation de frameworks comme [React Native](glossaire/react) permet, à travers un même code source, de produire des applications natives à destination des différentes plateformes. Ceci permet un gain de temps considérable, et offre des performances et une intégration à l'écosystème des différents appareils proche des applications développées en langages natifs.
+
+## La progressive Web App (PWA), une adaptation poussée de votre site web
+
+La **PWA** peut être définie comme un site internet, optimisé pour smartphone ou tablette, que vous pouvez installer de manière à y accéder rapidement depuis l'écran d'accueil, que vous soyez connecté ou non au réseau.
+La PWA va utiliser en sous-marin votre navigateur pour s'ouvrir et être parcourue, tout en vous proposant une expérience utilisateur (ou [UX](glossaire/ui-ux)) proche de celle de l'application native.
+
+Son gros avantage : elle n'a que faire de la marque ou du système d'exploitation de votre appareil, puisqu'elle exploite les fonctionnalités des navigateurs, comme _Firefox_, _Chrome_, _Edge_, _Safari_, etc. On parle alors d'**application multiplateforme**.
+
+Techniquement, on peut résumer la PWA à un site web, agrémenté d'une couche de configuration spécifique, notamment via la mise en place de _manifestes_, de _Service Workers_ et l'exploitation d'API de stockage local.
+Depuis la sortie de la version 16.4 d'iOS courant 2023, **une PWA peut recevoir des notifications web** sur ces appareils, ce qui la rapproche encore de l'expérience offerte par les applications natives.
+
+Il reste toutefois quelques limitations avec _Apple_, par exemple :
+- l'impossibilité d'ouvrir la PWA via un _deep link_, comme un lien dans un email, qui vous redirigera vers le site et non la PWA
+- le non partage de session qui ne permet pas d'être connecté automatiquement dans la PWA si vous l'étiez préalablement sur le site
+
+Certes marginales, ces menues différences peuvent, selon votre contexte, avoir leur importance. C'est pourquoi nous accompagnons et conseillons chacun de nos clients sur la solution la plus appropriée en fonction de ses enjeux, objectifs et moyens.
+
+## Comment choisir ?
+
+### Quand s'orienter vers une PWA ?
+
+- **Délai court de mise sur le marché** : Pour lancer rapidement un produit sans les délais des cycles d'approbation des app stores.
+- **Mises à jour fréquentes** : Si l'application nécessite des mises à jour régulières ou instantanées, une PWA est plus adaptée car elle se met à jour automatiquement à chaque visite. Toutefois, cet argument peut être contrebalancé par l'utilisation de certains outils comme _React Native_ et _Expo_ qui permettent des mises à jour sans passer par la validation des stores (_Over the Air updates_)
```

Comment:
> sans passer par la validation des stores (_Over the Air updates_)

Outre la validation des stores, cela permet surtout de s'affranchir de la nécessité pour l'utilisateur de mettre à jour l'application par lui-même ou de laisser l'OS le faire : il n'y a a tout moment qu'une seule version de l'application, à jour. 

Là où une application délivrée de façon plus classique mettra du temps avant que son parc d'utilisateur ait entièrement migré sur la dernière version (et ne sera même pas obligé de le faire).

Cela veut par exemple dire qu'il faut parfois gérer un versioning d'API pour supporter plusieurs versions de l'application la consommant.

Reply:
Oui c'est tout à fait vrai concernant Expo, mais la notion d'OTA (Over the Air updates) est faisable aussi autrement. Par exemple : https://github.com/microsoft/react-native-code-push